zerver/views/registration.py

```diff
@@ -515,3 +515,15 @@ def find_account(request: HttpRequest) -> HttpResponse:
                   'zerver/find_account.html',
                   context={'form': form, 'current_url': lambda: url,
                            'emails': emails},)
+
+def realm_redirect(request: HttpRequest) -> HttpResponse:
+    if request.method == 'POST':
+        form = RealmRedirectForm(request.POST)
+        if form.is_valid():
+            subdomain = form.cleaned_data['subdomain']
+            realm = get_realm(subdomain)
+            return redirect(urllib.parse.urljoin(realm.uri, request.GET.get("next", "")))
```

Comment:
I think we should be using `get_safe_redirect_to` here, so that the `/go/` URLs can't be used to maliciously create a URL going to a third-party site.